tests/test_immutable.py

```diff
@@ -60,21 +60,62 @@ def test_from_timeoffset(self):
+            (TimeOffset(1001, 1001.0/30000/2*1000000000 + 1), Fraction(30000, 1001), TimeOffset.ROUND_NEAREST,
+             TimeOffset(1001, 1001.0/30000*1000000000)),
+            (TimeOffset(1001, 1001.0/30000/2*1000000000, -1), Fraction(30000, 1001), TimeOffset.ROUND_NEAREST,
+             TimeOffset(1001, 0, -1)),
+            (TimeOffset(1001, 1001.0/30000/2*1000000000 + 1, -1), Fraction(30000, 1001), TimeOffset.ROUND_NEAREST,
+             TimeOffset(1001, 1001.0/30000*1000000000, -1)),
+            (TimeOffset(1521731233, 320000000), Fraction(25, 3), TimeOffset.ROUND_NEAREST,
+             TimeOffset(1521731233, 320000000)),
+            (TimeOffset(0, 0), Fraction(30000, 1001), TimeOffset.ROUND_UP,
+             TimeOffset(0, 0)),
+            (TimeOffset(1001, 0), Fraction(30000, 1001), TimeOffset.ROUND_UP,
+             TimeOffset(1001, 0)),
+            (TimeOffset(1001, 1001.0/30000/2*1000000000), Fraction(30000, 1001), TimeOffset.ROUND_UP,
+             TimeOffset(1001, 1001.0/30000*1000000000)),
+            (TimeOffset(1001, 1001.0/30000/2*1000000000 + 1), Fraction(30000, 1001), TimeOffset.ROUND_UP,
+             TimeOffset(1001, 1001.0/30000*1000000000)),
+            (TimeOffset(1001, 1001.0/30000/2*1000000000, -1), Fraction(30000, 1001), TimeOffset.ROUND_UP,
+             TimeOffset(1001, 0, -1)),
+            (TimeOffset(1001, 1001.0/30000/2*1000000000 + 1, -1), Fraction(30000, 1001), TimeOffset.ROUND_UP,
+             TimeOffset(1001, 0, -1)),
+            (TimeOffset(1521731233, 320000000), Fraction(25, 3), TimeOffset.ROUND_UP,
+             TimeOffset(1521731233, 320000000)),
+            (TimeOffset(0, 0), Fraction(30000, 1001), TimeOffset.ROUND_DOWN,
+             TimeOffset(0, 0)),
+            (TimeOffset(1001, 0), Fraction(30000, 1001), TimeOffset.ROUND_DOWN,
+             TimeOffset(1001, 0)),
+            (TimeOffset(1001, 1001.0/30000/2*1000000000), Fraction(30000, 1001), TimeOffset.ROUND_DOWN,
+             TimeOffset(1001, 0)),
+            (TimeOffset(1001, 1001.0/30000/2*1000000000 + 1), Fraction(30000, 1001), TimeOffset.ROUND_DOWN,
+             TimeOffset(1001, 0)),
+            (TimeOffset(1001, 1001.0/30000/2*1000000000, -1), Fraction(30000, 1001), TimeOffset.ROUND_DOWN,
+             TimeOffset(1001, 1001.0/30000*1000000000, -1)),
+            (TimeOffset(1001, 1001.0/30000/2*1000000000 + 1, -1), Fraction(30000, 1001), TimeOffset.ROUND_DOWN,
+             TimeOffset(1001, 1001.0/30000*1000000000, -1)),
+            (TimeOffset(1521731233, 320000000), Fraction(25, 3), TimeOffset.ROUND_DOWN,
+             TimeOffset(1521731233, 320000000)),
         ]
 
-        for t in tests_ts:
-            with self.subTest(t=t):
-                r = t[0].normalise(t[1][0], t[1][1])
-                self.assertEqual(r, t[2],
-                                 msg=("{!r}.normalise({}, {}) == {!r}, expected {!r}"
-                                      .format(t[0], t[1][0], t[1][1], r, t[2])))
+        n = 0
```

Comment:
What does `n` do?

Reply:
Keep track of which iteration you're on, so that it can be included in the subTest information below

Reply:
Should we give a more meaningful name like `iterationNumber`/`iteration_number`/`iteration`?